### PR TITLE
build(node): Upgrade to nodejs 18

### PR DESCRIPTION
**Pull request by gidjin**
# Summary

Upgrades node from 16.x to 18.x since 16 is EOL 2023-09-11 See [nodejs release schedule](https://github.com/nodejs/release#release-schedule) and [update about the EOL of node 16.x](https://nodejs.org/en/blog/announcements/nodejs16-eol/)

Also drops support for unsupported versions of node 10.x - 12.x

## Related Issues or PRs

#2129 

## How To Test

Ensure you can install dependencies, run various commands like `yarn lint`, `yarn test`, and `yarn storybook`.

### CODE REVIEW

**Review by brandonlenz — APPROVED** [review]
@gidjin there's one more reference to node 16.13.0 in .circleci.config.yml that I think we might as well update too.

Approving to not be a blocker on it

**Comment by gidjin** [comment]
> @gidjin there's one more reference to node 16.13.0 in .circleci.config.yml that I think we might as well update too.
> 
> Approving to not be a blocker on it

Thanks for catching that! Fun fact our configuration requires re-review of approved PRs that are updated.

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ jobs:
       - auth:
           password: $DOCKER_PASSWORD
           username: $DOCKER_USERNAME
-        image: circleci/node:16.13.0
+        image: circleci/node:18.10.0
```
Comment (brandonlenz) [inline .circleci/config.yml:8]: Looks like the build failed. I'm wondering if we need to use the latest? Looks like we're attempting to use a deprecated image https://circleci.com/developer/images/image/cimg/node

Comment (brandonlenz) [reply .circleci/config.yml:8]: It looks like the deprecated circleci/node:* images don't have one for this node version based on the build failure we got https://app.circleci.com/pipelines/github/trussworks/react-uswds/9313/workflows/5d04dc51-0c15-4d6c-ae43-45911c6963d3/jobs/9348/steps

Comment (brandonlenz) [reply .circleci/config.yml:8]:
```suggestion
        image: cimg/node:18.10.0
```

Comment (gidjin) [reply .circleci/config.yml]: Just to note, this new image is the one to use, but it has also removed `python` version 2.x so I had to update the deploy scripts to use `python3` instead.

**Review by brandonlenz — APPROVED** [review]

**Pull request merged**